---
layout: single
title: Overflow of Int in Python
---

Solving problems on Project Euler yesterday, I find that python has its own big number limit, when using numpy.

Here is what I wrote when I am solving Fibonacci sequence using matrix acceleration.

```python
fib = np.ones((2, 2), int)
fib[1][1] = 0
```

When I print out

```python
print(fib.dtype)
```

It gives

```
int64
```

So numpy will automatically interpret `int` to `int64`(up to the machine, possibly `int32`), as direct operations without numpy won't be affected.

```python
a = 1234567891011 # larger than 2^32-1
b = 1101987654321
c = a * b
print(c)
```

It gives 

```
1360478574315235871208531
```



Here is the solution given by chatGPT/

```python
fib = np.ones((2, 2), dtype=object)
```

And it runs well.

